### PR TITLE
[Merged by Bors] - feat: `Algebra.adjoin R s` is `span R {1} ⊔ s.toSubmodule`

### PR DESCRIPTION
**Pull request by j-loreaux**
This relates several unital and non-unital subobject closures to each other the to various counterparts.

This is one of the key steps towards establishing that `C(s, 𝕜)₀` is the closure of the non-unital star subalgebra generated by the identity function, when `s` is compact and `RCLike 𝕜`. This, in turn, is essential for writing instances of the continuous functional calculus for non-unital algebras.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Group/Submonoid/Basic.lean [hunk 1]
```diff
@@ -489,6 +489,23 @@ theorem dense_induction {p : M → Prop} (x : M) {s : Set M} (hs : closure s = 
 #align submonoid.dense_induction Submonoid.dense_induction
 #align add_submonoid.dense_induction AddSubmonoid.dense_induction
 
+/-- The `Submonoid.closure` of a set is `1` union the its `Subsemigroup.closure`. -/
+lemma closure_eq_one_union (s : Set M) :
```
Comment (dupuisf) [inline Mathlib/Algebra/Group/Submonoid/Basic.lean:493]:
```suggestion
/-- The `Submonoid.closure` of a set is the union of `{1}` and its `Subsemigroup.closure`. -/
lemma closure_eq_one_union (s : Set M) :
```

File: Mathlib/Algebra/Star/NonUnitalSubalgebra.lean [hunk 2]
```diff
@@ -806,6 +831,15 @@ variable [Module R B] [IsScalarTower R B B] [SMulCommClass R B B] [StarModule R
 variable [FunLike F A B] [NonUnitalAlgHomClass F R A B] [NonUnitalStarAlgHomClass F R A B]
 variable (S : NonUnitalStarSubalgebra R A)
 
+lemma _root_.NonUnitalStarAlgHom.map_adjoin (f : F) (s : Set A) :
+    map f (adjoin R s) = adjoin R (f '' s) :=
+  Set.image_preimage.l_comm_of_u_comm (gc_map_comap f) NonUnitalStarAlgebra.gi.gc
+    NonUnitalStarAlgebra.gi.gc fun _t => rfl
+
+@[simp]
+lemma map_adjoin_singleton (f : F) (x : A) : map f (adjoin R {x}) = adjoin R {f x} := by
```
Comment (dupuisf) [inline Mathlib/Algebra/Star/NonUnitalSubalgebra.lean:840]:
```suggestion
@[simp]
lemma _root_.NonUnitalStarAlgHom.map_adjoin_singleton (f : F) (x : A) : map f (adjoin R {x}) = adjoin R {f x} := by
```
(i.e. as above?)

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors d+

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**